examples/AcousticWave2D_codegen.py

```diff
@@ -95,21 +95,43 @@ def _init_taylor(self, nt):
         # domain to the border
 
         # Forward wave equation
+
+        def damp_boundary(damp):
+            h = self.h
+            dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40 * h)
+            nx, ny = self.model.get_shape()
+            nbpml = self.nbpml
+            for i in range(nbpml):
+                pos = np.abs((nbpml-i)/nbpml)
+                val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
+                damp[i, :] += val
+                damp[-i, :] += val
+                damp[:, i] += val
+                damp[:, -i] += val
+
         wave_equation = m * dtt - (dxx + dyy) + e * dt
         stencil = solve(wave_equation, p(x, y, t+s))[0]
         self.nt = nt
-
+        m_sub = DenseData("m", self.model.vp.shape, self.dtype)
+        m_sub.initializer = lambda ref: np.copyto(ref, self.model.vp**(-2))
```

Comment:
I am not defending deferred initialisation. As I said before, I am happy to remove the deferred initialisation. What should it be replaced with? That's my question. 

I see the ``**-2`` operation as a preprocessing step that actually belongs in the place where the model data is being initialised - just like the rest of the initialisation, there is no need to delay this. So, to me, ``self.model.vp**-2`` is just ``self.model.vp`` where the ``**2`` operation was already done in the initialisation of the model. @mloubout or @ggorman can comment better on this. 

So, assuming the ``self.model.vp**-2`` can (and should be) just be replaced by a ``self.model.vp`` very easily, are you saying that ``m_sub.data = self.model.vp`` does any more than pointer assignment?

So where do you see the premature optimisation and over-engineering?